packages/core/contracts/financial-templates/common/FeePayer.sol

```diff
@@ -148,6 +145,27 @@ abstract contract FeePayer is AdministrateeInterface, Testable, Lockable {
         return totalPaid;
     }
 
+    function getOutstandingRegularFees(uint256 time)
+        public
+        view
+        returns (FixedPoint.Unsigned memory regularFee, FixedPoint.Unsigned memory latePenalty)
+    {
+        StoreInterface store = _getStore();
+        FixedPoint.Unsigned memory collateralPool = _pfc();
+
+        // Exit early if there is no collateral from which to pay fees.
+        if (collateralPool.isEqual(0)) {
+            return (regularFee, latePenalty);
+        }
+
+        // Exit early if fees were already paid during this block.
+        if (lastPaymentTime == time) {
+            return (regularFee, latePenalty);
+        }
+
```

Comment:
nit: don't we also need to move the block of code that starts with:
```
if (totalPaid.isGreaterThan(collateralPool)) {
```

into this function in case the fees are > pfc?

Reply:
Good point. this check & the associated logic needs to be put into the `getOutstandingRegularFees` method.

I have refactored this pretty drastically to accommodate this and clean up this implementation. PTAL